test/__snapshots__/step-function.test.ts.snap

```diff
@@ -7483,6 +7511,85 @@ Object {
 }
 `;
 
+exports[`stringify json 1`] = `
+Object {
+  "StartAt": "Initialize Functionless Context",
+  "States": Object {
+    "1__return JSON.stringify(input)": Object {
+      "End": true,
+      "InputPath": "$.heap0.string",
+      "ResultPath": "$",
+      "Type": "Pass",
+    },
+    "Initialize Functionless Context": Object {
+      "Next": "return JSON.stringify(input)",
+      "OutputPath": "$",
+      "Parameters": Object {
+        "input.$": "$",
+        "null": null,
+      },
+      "ResultPath": "$.fnl_context",
+      "Type": "Pass",
+    },
+    "return JSON.stringify(input)": Object {
+      "Next": "1__return JSON.stringify(input)",
+      "Parameters": Object {
+        "string.$": "States.jsonToString($.fnl_context.input)",
+      },
+      "ResultPath": "$.heap0",
+      "Type": "Pass",
+    },
+  },
+}
+`;
+
+exports[`stringify object literal 1`] = `
+Object {
+  "StartAt": "return JSON.stringify({a: \\"a\\", b: {c: \\"c\\"}})",
+  "States": Object {
+    "1__return JSON.stringify({a: \\"a\\", b: {c: \\"c\\"}})": Object {
+      "End": true,
+      "InputPath": "$.heap0.string",
+      "ResultPath": "$",
+      "Type": "Pass",
+    },
+    "1__return JSON.stringify({a: \\"a\\", b: {c: \\"c\\"}}) 1": Object {
+      "Next": "1__return JSON.stringify({a: \\"a\\", b: {c: \\"c\\"}})",
+      "Parameters": Object {
+        "string.$": "States.jsonToString($.heap1)",
+      },
+      "ResultPath": "$.heap0",
+      "Type": "Pass",
+    },
+    "return JSON.stringify({a: \\"a\\", b: {c: \\"c\\"}})": Object {
+      "Next": "1__return JSON.stringify({a: \\"a\\", b: {c: \\"c\\"}}) 1",
+      "Result": Object {
+        "a": "a",
+        "b": Object {
+          "c": "c",
+        },
+      },
+      "ResultPath": "$.heap1",
+      "Type": "Pass",
+    },
+  },
+}
+`;
+
+exports[`stringify undefined 2 1`] = `
+Object {
+  "StartAt": "return JSON.stringify(undefined)",
+  "States": Object {
+    "return JSON.stringify(undefined)": Object {
+      "End": true,
+      "InputPath": "$.heap0",
+      "ResultPath": "$",
+      "Type": "Pass",
+    },
```

Comment:
What does this output? I suspect it will be an error or `null`?

Reply:
This would be a validation error, but its tough to catch without having types at transpile time. 

```ts
return JSON.stringify(undefined) // fails
return JSON.stringify(undefined) ?? null // succeeds
```